### PR TITLE
Remove pygments and add chroma

### PR DESCRIPTION
**Pull request by joshcarp**
- Removes vendor directory and uses chroma syntax highlighting instead
[github.com/alecthomas/chroma]()
- Add gomod 
- Generates files with new generate.go
- Syntax highlighting seems to be slightly different
I found this greatly simplifies the code dependencies

### CODE REVIEW

**Comment by mmcgrana** [comment]
Interesting suggestion, thanks @Joshcarp!

I'm open to this change. To consider merging it we'd need to:

- Update README
- Update Travis script
- Fix highlighting so that it's the same as the current site (which may just be a matter of styling?)

LMK if you're interested in taking that on.

**Comment by eliben** [comment]
Coincidentally, I notice this PR updates BlackFriday to 2.0, which might help resolve #182 
It was waiting for BlackFriday to start supporting go modules, and it seems like they do now

File: tools/generate.go [hunk 1]
```diff
@@ -216,26 +188,48 @@ func parseSegs(sourcePath string) ([]*Seg, string) {
...
 	segs, filecontent := parseSegs(sourcePath)
-	lexer := whichLexer(sourcePath)
+	var goFile bool
 	for _, seg := range segs {
 		if seg.Docs != "" {
 			seg.DocsRendered = markdown(seg.Docs)
 		}
 		if seg.Code != "" {
-			seg.CodeRendered = cachedPygmentize(lexer, seg.Code)
+			seg.CodeRendered = cachedPygmentize(sourcePath, seg.Code)
```
Comment (eliben) [inline tools/generate.go]: Maybe this should be renamed now to avoid "pygment*" ?

File: tools/generate.go [hunk 2]
```diff
@@ -216,26 +188,48 @@ func parseSegs(sourcePath string) ([]*Seg, string) {
...
 	}
 	// we are only interested in the 'go' code to pass to play.golang.org
-	if lexer != "go" {
+	if !goFile {
 		filecontent = ""
 	}
 	return segs, filecontent
 }
 
+func chromaFormat(code, fileExtension string) string {
```
Comment (eliben) [inline tools/generate.go]: It's a bit confusing that the second arg is called `fileExtension` when you're passing in the whole path. Can you please add a comment to this function and align the arg name with usage?

File: tools/generate.go [hunk 3]
```diff
@@ -216,26 +188,48 @@ func parseSegs(sourcePath string) ([]*Seg, string) {
...
+	lexer := lexers.Get(fileExtension)
+	if lexer == nil {
+		lexer = lexers.Fallback
```
Comment (eliben) [inline tools/generate.go]: Is this actually needed? We're only formatting .go files and .sh files, right?

**Comment by joshcarp** [comment]
Do you find it useful?
Sort of forgot about this one...

**Comment by eliben** [comment]
> Do you find it useful?
> Sort of forgot about this one...

Yes, see Mark's reply in https://github.com/mmcgrana/gobyexample/pull/306#issuecomment-570804157 and also my review comments.

**Pull request closed without merging**